consensus/istanbul/core/handler.go

```diff
@@ -32,8 +32,23 @@ func (c *core) ParentCommits() MessageSet {
 
 // Start implements core.Engine.Start
 func (c *core) Start() error {
-	// Start a new round from last sequence + 1
-	c.startNewRound(common.Big0)
+	headBlock, headAuthor := c.backend.GetCurrentHeadBlockAndAuthor()
+	valSet := c.backend.Validators(headBlock)
+
+	nextBlock := new(big.Int).Add(headBlock.Number(), common.Big1)
+	nextProposer := c.selectProposer(valSet, headAuthor, 0)
```

Comment:
What if we shut down in the middle sequence N, and restart when the network is still on sequence N? Wouldn't we want to load our latest round state?

Reply:
On line 40, we call `newRoundStateWithPersistence()` which will check which is the last saved View. If the stored view >= the computed view here, we will ignore all the parameters and use the stored version.

So, in essence calling `newRoundStateWithPersistence()` is like saying "try restoring a newer version of roundState and if not, create a new one with this parameters"

Reply:
I've changed to:

```go
	roundState, err := c.createRoundState()
	if err != nil {
		return err
	}

	c.current = roundState
	c.roundChangeSet = newRoundChangeSet(c.current.ValidatorSet())
```

Also, fixed a bug where i was using a maybe old valSet to create the `RoundChangeSet` :D